client2/thin/thin.go

```diff
@@ -280,6 +280,8 @@ func (t *ThinClient) worker() {
 		if err != nil {
 			t.log.Errorf("thin client ReceiveMessage failed: %v", err)
 			if err == io.EOF {
+				// XXX: should we halt ThinClient on EOF??
```

Comment:
yes i think we should halt the thin client on EOF... and we should remove that comment and ideally not put them in our code